### PR TITLE
Implement more bindings for pelikan

### PR DESCRIPTION
**Pull request by swlynch99**
This has been pulled out as a part of #276. It's mostly just implementing bindings, various traits, etc for the rest of the code.

### CODE REVIEW

File: src/rust-util/pelikan-sys/Cargo.toml [hunk 1]
```diff
@@ -10,9 +10,11 @@ cdb = [ "cdb_rs" ]
 client-network = []
 core = []
 cuckoo = []
+datapool = []
 ds_bitmap = []
 ds_sarray = []
 ds_ziplist = []
+hotkey = [ "datapool"]
```
Comment (brayniac) [inline src/rust-util/pelikan-sys/Cargo.toml]: add whitespace before closing `]`

File: src/rust-util/pelikan-sys/Cargo.toml [hunk 2]
```diff
@@ -22,10 +24,13 @@ slab = []
 time = []
 util = [ "time" ]
 
-default = [ "core", "time", "util" ]
+default = [ 
+    "core", "time", "util",
```
Comment (brayniac) [inline src/rust-util/pelikan-sys/Cargo.toml]: Any particular reason for this formatting? I'd prefer one line, or line per feature here.

File: src/rust-util/pelikan-sys/build.rs [hunk 3]
```diff
@@ -44,12 +44,30 @@ fn builder() -> bindgen::Builder {
 
     builder
         .constified_enum(".*")
-        .whitelist_recursively(false)
+        // .whitelist_recursively(false)
```
Comment (brayniac) [inline src/rust-util/pelikan-sys/build.rs]: remove this line instead of leaving it commented

File: src/rust-util/pelikan-sys/Cargo.toml (hunk 2 above)
Comment (swlynch99) [reply src/rust-util/pelikan-sys/Cargo.toml]: This was left over from when I added some other default features temporarily. I changed it to be all on one line.

**Pull request merged**